### PR TITLE
[prometheus] Make Prometheus configmap reload can have custom reload url

### LINKED ISSUE #3295: [prometheus] Make Prometheus configmap reload can have custom reload url
### Is your feature request related to a problem ?

I want to enable basic authentication for Prometheus web site, but currently the configMap reload url is hard-coded to `--reload-url=http://127.0.0.1:9090{{ .Values.server.prefixURL }}/-/reload`.

### Describe the solution you'd like.

Make that reload url be able to override by the values file.

### Describe alternatives you've considered.

NONE.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by nguyenhoaibao**
#### What this PR does / why we need it

So we can custom the reload url to make it work with basic authentication, for example.

#### Which issue this PR fixes

- fixes #3295 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/sts.yaml [hunk 1]
```diff
@@ -65,7 +65,11 @@ spec:
           {{- end }}
           args:
             - --watched-dir=/etc/config
+          {{- if .Values.configmapReload.reloadUrl }}
+            - --reload-url={{ .Values.configmapReload.reloadUrl }}
+          {{- else }}
             - --reload-url=http://127.0.0.1:9090{{ .Values.server.prefixURL }}/-/reload
+          {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus/templates/sts.yaml:72]:
```suggestion
          {{- $default_url := printf "http://127.0.0.1:9090/%s/-/reload" .Values.server.prefixURL }}
          - --reload-url={{ default $default_url .Values.configmapReload.reloadUrl }}
```

File: charts/prometheus/templates/deploy.yaml [hunk 2]
```diff
@@ -65,7 +65,11 @@ spec:
           {{- end }}
           args:
             - --watched-dir=/etc/config
+          {{- if .Values.configmapReload.reloadUrl }}
+            - --reload-url={{ .Values.configmapReload.reloadUrl }}
+          {{- else }}
             - --reload-url=http://127.0.0.1:9090{{ .Values.server.prefixURL }}/-/reload
+          {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus/templates/deploy.yaml:72]:
```suggestion
          {{- $default_url := printf "http://127.0.0.1:9090/%s/-/reload" .Values.server.prefixURL }}
          - --reload-url={{ default $default_url .Values.configmapReload.reloadUrl }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus/templates/sts.yaml (hunk 1 above)
Comment (nguyenhoaibao) [reply charts/prometheus/templates/sts.yaml]: Done

File: charts/prometheus/templates/deploy.yaml (hunk 2 above)
Comment (nguyenhoaibao) [reply charts/prometheus/templates/deploy.yaml]: Done

**Review by zanhsieh — APPROVED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**